include/LayerPlan.h

```diff
@@ -820,6 +835,114 @@ class LayerPlan : public NoCopy
         const coord_t wipe_dist,
         const Ratio flow_ratio,
         const double fan_speed);
+
+    /*!
+     *  @brief Send a GCodePath line to the communication object, applying proper Z offsets
+     *  @param path The path to be sent
+     *  @param position The start position (which is not included in the path points)
+     *  @param extrude_speed The actual used extrusion speed
+     */
+    void sendLineTo(const GCodePath& path, const Point3LL& position, const double extrude_speed);
+
+    /*!
+     *  @brief Write a travel move and properly apply the various Z offsets
+     *  @param gcode The actual GCode exporter
+     *  @param position The position to move to. The Z coordinate is an offset to the current layer position
+     *  @param speed The actual used speed
+     *  @param path_z_offset The global path Z offset to be applied
+     *  @note This function is to be used when dealing with 3D coordinates. If you have 2D coordinates, just call gcode.writeTravel()
+     */
+    void writeTravelRelativeZ(GCodeExport& gcode, const Point3LL& position, const Velocity& speed, const coord_t path_z_offset);
+
+    /*!
+     * \brief Write an extrusion move and properly apply the various Z offsets
+     * \param gcode The actual GCode exporter
+     * \param position The position to move to. The Z coordinate is an offset to the current layer position
+     * \param speed The actual used speed
+     * \param path_z_offset The global path Z offset to be applied
+     * \param extrusion_mm3_per_mm The desired flow rate
+     * \param feature The current feature being printed
+     * \param update_extrusion_offset whether to update the extrusion offset to match the current flow rate
+     */
+    void writeExtrusionRelativeZ(
+        GCodeExport& gcode,
+        const Point3LL& position,
+        const Velocity& speed,
+        const coord_t path_z_offset,
+        double extrusion_mm3_per_mm,
+        PrintFeatureType feature,
+        bool update_extrusion_offset = false);
+
+    /*!
+     * \brief Add a wall to the gcode with optimized order
```

Comment:
I would have expected a bit more information how this "splitWall" is different from any other. I'm not entirely sure what the split means in this context.

Reply:
Done!